main.py

```diff
@@ -189,6 +191,8 @@ def check_switch_consistency(self, switch):
         serializer = FlowFactory.get_class(switch)
 
         for stored_flow in stored_flows:
+            if time.time() - stored_flow.get('timestamp', 0) < STATS_INTERVAL:
```

Comment:
@italovalcy I wonder if we could store this as a iso 8601 str, just so it would be immediately easier for network operators or developers to tell how old or new an entry is if they're analyzing the stored data structure. I believe it would also be slightly easier to convert and interface with future DBs when trying to set this value as `datetime` and leverage more powerful `datetime` aware queries one day when the time comes. Wdyt?


```
In [1]: from datetime import datetime

In [2]: from datetime import timezone

In [3]: t1 = datetime.now(timezone.utc).isoformat()

In [4]: t1
Out[4]: '2021-11-03T20:27:02.073618+00:00'

In [5]: diff = (datetime.now(timezone.utc) - datetime.fromisoformat(t1)).seconds

In [6]: diff
Out[6]: 14

In [7]: diff = (datetime.now(timezone.utc) - datetime.fromisoformat(t1)).seconds

In [8]: diff
Out[8]: 40

```

Reply:
Hi @viniarck thank you for your suggestion. I agree that using ISO string format would be easier to read, however, I'm just not sure this will perform well. Instead of storing ISO string format, we could keep the float `time.time()` and do some transformations when someone requests via the GET API. I've performed a very poor/basic benchmark to compare both formats (1), and it turns out using `time.time()` is almost 4x faster to calculate the diff and it saves almost 3x memory space.

(1) basic experiment comparing `time.time()` versus `datetime` string representation. Code:

```
from datetime import datetime
from datetime import timezone
import time

t1 = datetime.now(timezone.utc).isoformat()
for i in range(1, 100000):
    x = (datetime.now(timezone.utc) - datetime.fromisoformat(t1)).seconds
print((datetime.now(timezone.utc) - datetime.fromisoformat(t1)).total_seconds())

t1 = time.time()
for i in range(1, 100000):
    x = time.time() - t1
print(time.time() - t1)
```

The experiment was repeated 10 times, and the following are the results considering the average and confidence interval of 95%: datetime string format took 64.7 +- 2.6 ms to process 100k diff calculations, while the float diff from time.time() took 13.9 +- 0.2 ms. In terms of space complexity, the string representation consumes 68 bytes, while the float representation 24 bytes.

Having said that, commit 4d826cb changes the way the timestamp is stored to meet the suggestion! In the future, we can revisit this discussion.

Reply:
thanks @italovalcy for pushing this commit, and also performing benchmarks, yeah for sure, let's keep this in mind and revisit when we also start using other DB backends other than the FS one.